src/prototype.rs

```diff
@@ -1035,7 +1167,7 @@ fn get_html_end_tag(text : &str) -> Option<&'static str> {
     static END_TAGS: &'static [&'static str; 3] = &["</script>", "</pre>", "</style>"];
 
     for (beg_tag, end_tag) in BEGIN_TAGS.iter().zip(END_TAGS.iter()) {
-        if 1 + beg_tag.len() < text.len() &&
+        if 1 + beg_tag.len() <= text.len() &&
```

Comment:
I think this length check (and the one below) should be removed. According to the spec, the start condition is satisfied if the line is just the begin tag and a newline.

Also, spec says matching is case insensitive, so `starts_with` isn't quite right.

Reply:
> spec says matching is case insensitive,

Something like `text[..beg_tag.len()].to_lowercase().starts_with()` ? In which case we do need the length check after all? Is there a better way to do that?

Reply:
Feel free to leave this as a TODO. That code snippet is almost correct but can panic if the tag length slices a Unicode codepoint in half. It also allocates a String which is not great for performance. Doing this fully correctly in the Unicode case is tricky, but fortunately the actual logic only needs ASCII. This is much simpler, but I'm not sure that function exists in the Rust std lib.

Also, on consulting the spec, I don't think the condition is getting it exactly right; it should check that there's whitespace etc after the tag name, so for example will reject `<prepare>`.

Reply:
Thanks for the suggestion, I tried something different in the latest update.